Net/src/SocketImpl.cpp

```diff
@@ -1433,21 +1434,26 @@ Poco::Int64 _sendfile(poco_socket_t sd, FileIOS::NativeHandle fd, Poco::UInt64 o
 	return sent;
 }
 
-Poco::Int64 SocketImpl::sendFile(FileInputStream &fileInputStream, Poco::UInt64 offset)
+IntPtr SocketImpl::sendFile(FileInputStream &fileInputStream, UIntPtr offset)
 {
 	FileIOS::NativeHandle fd = fileInputStream.nativeHandle();
-	Poco::UInt64 fileSize = fileInputStream.size();
+	UIntPtr fileSize = fileInputStream.size();
 	std::streamoff sentSize = fileSize - offset;
-	Poco::Int64 sent = 0;
+	IntPtr sent = 0;
```

Comment:
Still if _sendFile sends multiple times this will be just last sent. So user will not actually get total size which is sended just last send size.

Reply:
No, sendFile sends only once, it repeat only when nothing was sent, it can happen when errno is equal to EAGAIN or EWOULDBLOCK